js_modules/dagster-ui/packages/ui-core/src/nav/LeftNavItem.tsx

```diff
@@ -114,20 +135,83 @@ export const LeftNavItem = React.forwardRef(
           ) : null}
         </>
       );
-    };
+    }, [repoAddress, schedules, sensors, showDialog, status]);
+
+    const setFragmentData = React.useCallback((fragment: InstigationStateFragment) => {
+      setData((data) => {
+        return {
+          ...data,
+          [fragment.id]: fragment,
+        };
+      });
+    }, []);
+
+    /**
+     * This is pretty clowny but `INSTIGATION_STATES_QUERY` only returns instigation states for sensors/schedules that were  previously turned on or off.
```

Comment:
Took me a minute to grok this, I appreciate the detailed comment! 

I feel like we should push pretty hard on the backend to change this - or at least do this status fallback lookup at the graphql layer? It's a bummer to have this much complexity in the front-end :(

Reply:
yeah agreed. Doing the fallback lookup on the server would make the query slow again though :(